### PR TITLE
STITCH-2742: Update release scripts and README to reflect new Github requirements

### PR DESCRIPTION
**Pull request by tkaye407**

### CODE REVIEW

**Comment by tkaye407** [comment]
I used a tool called hub to open the pull request because git request-pull doesn't have the desired behaviour: https://stackoverflow.com/questions/34945947/git-request-pull-how-to-create-a-github-pull-request-on-the-command-line

File: contrib/README.md [hunk 1]
```diff
@@ -6,12 +6,13 @@ This project follows [Semantic Versioning 2.0](https://semver.org/). In general,
...
 ```bash
-# run bump_version.bash with either patch, minor, or major
-./bump_version.bash <snapshot|beta|patch|minor|major>
-
-# send an email detailing the changes to the https://groups.google.com/d/forum/mongodb-stitch-announce mailing list
+# run bump_version.bash with either patch, minor, or major followed by the JIRA ticket number (you may omit the STITCH keyword if you would like).
+./bump_version.bash <snapshot|beta|patch|minor|major> <STITCH-1234|1234>
 ```
 
+* send an email detailing the changes to the https://groups.google.com/d/forum/mongodb-stitch-announce mailing list
```
Comment (adamchel) [inline contrib/README.md]: [nit] we no longer do this. replace this with:

```
*  Publish a release for the new SDK version on the GitHub repository and 
   include relevant release notes. See
   https://help.github.com/en/articles/creating-releases for context, and 
   follow the general format of our previous releases.
```

File: contrib/README.md [hunk 2]
```diff
@@ -23,6 +24,9 @@ publish.bintray.mavenSyncUser=<maven_central_sync_user> # optional
 publish.bintray.mavenSyncPassword=<maven_central_sync_password> # optional
 ```
 
+#### Configuring Hub
+For the `./gradlew bintrayUpload` command to work properly, you must have ```hub``` installed. Please see [Hub](https://github.com/github/hub) for installation details.
```
Comment (adamchel) [inline contrib/README.md]: [nit] for `bump_version.bash` to work properly

File: contrib/README.md [hunk 3]
```diff
@@ -6,12 +6,13 @@ This project follows [Semantic Versioning 2.0](https://semver.org/). In general,
...
-# run bump_version.bash with either patch, minor, or major
-./bump_version.bash <snapshot|beta|patch|minor|major>
-
-# send an email detailing the changes to the https://groups.google.com/d/forum/mongodb-stitch-announce mailing list
+# run bump_version.bash with either patch, minor, or major followed by the JIRA ticket number (you may omit the STITCH keyword if you would like).
+./bump_version.bash <snapshot|beta|patch|minor|major> <STITCH-1234|1234>
 ```
 
+* send an email detailing the changes to the https://groups.google.com/d/forum/mongodb-stitch-announce mailing list
+* go to [Android SDK](https://github.com/mongodb/stitch-android-sdk/pulls) and request a reviewer on the pull request (mandatory) before merging and deleting the release branch
```
Comment (adamchel) [inline contrib/README.md]: [q] does `bump_version.bash` not do this automatically with the new `hub` command?

File: contrib/bump_version.bash [hunk 4]
```diff
@@ -85,16 +98,19 @@ then
...
+    set +e
+    git commit -m "$JIRA_TICKET: Update $NEW_VERSION"
+    if [ $? -eq 0 ]; then
+	    echo "creating pull request in github..."
+        set -e
+        git push -u origin "Release-$NEW_VERSION"
+        hub pull-request -m "$JIRA_TICKET: Release $NEW_VERSION" --base mongodb:master --head mongodb:"Release-$NEW_VERSION"
 	fi
 	set -e
 fi
```
Comment (adamchel) [inline contrib/bump_version.bash]: [minor] I think everything below this line should be in a separate script (`contirb/publish_sdk.sh`) that should only be run after the pull request is merged. We shouldn't be publishing anything to bintray before it hits master. When you move this stuff to a new script, the $NEW_VERSION variables will have to be replaced with the current version, which you can fetch the same way as how you fetch LAST_VERSION in this script. ask me in person if you have any questions

**Pull request closed without merging**